ev3dev/helper.py

```diff
@@ -311,28 +311,56 @@ def __init__(self, motor_specs, desc=None):
         (self.left_motor, self.right_motor) = self.motors.values()
         self.max_speed = self.left_motor.max_speed
 
-    def set_speed_ratio(self, left_motor_ratio, right_motor_ratio, set_right=True):
+    def set_speed_steering(self, direction, power=100):
         """
-        Set the speeds of the left_motor vs right_motor by ratio
+        Set the speed_sp for each motor in a pair to achieve the specified steering
+
+        direction [-100, 100]:
+            * -100 means turn left as fast as possible,
+            *  0   means drive in a straight line, and
+            *  100 means turn right as fast as possible.
+
+        power: the power that should be applied to the outmost motor (the one
+               rotating faster). The power of the other motor will be computed
+               automatically.
         """
 
-        # Set the right_motor speed relative to the left_motor_speed
-        if set_right:
-            self.right_motor.speed_sp = int((self.left_motor.speed_sp * left_motor_ratio) / right_motor_ratio)
+        assert direction >= -100 and direction <= 100,\
+            "%s is an invalid direction, must be between -100 and 100 (inclusive)" % direction
+
+        left_power = power
+        right_power = power
+        speed = (50 - abs(float(direction))) / 50
+
+        if direction >= 0:
+            right_power *= speed
         else:
-            self.left_motor.speed_sp = int((self.right_motor.speed_sp * right_motor_ratio) / left_motor_ratio)
+            left_power *= speed
+
+        left_power = int(left_power)
+        right_power = int(right_power)
+        log.debug("%s: direction %d, power %d, speed %d, left_power %d, right_power %d" %
+            (self, direction, power, speed, left_power, right_power))
+
+        self.left_motor.speed_sp = int(left_power)
```

Comment:
We don't actually run a command here; is the expectation that you call `set_speed_steering` and then set the command in a loop? Note that updating the setpoint won't have an effect until a command is sent AFAIK.

Reply:
yep it will be up to the user to call run_forever, run_timed, etc after this....all this function does is set them up with the desired speed_sp setting.  Same goes for set_speed_percentage().

Reply:
Hmm... In that case, we need to make sure we document the behavior!